File: linked_lists/stack.java
class Stack<T> {
    class Node{
        public Node next;
        public T data ;
        public Node(T data, Node next){
            this.data  = data ;
            this.next = next ;
        }
    }
    public Node head;


    public Stack(){
        head = null;
    }
    public void push(T data){
        head = new Node(data, this.head);
    }
    public T pop() throws Exception {
        if(this.head != null ){
            T data = this.head.data ;
            this.head = this.head.next;
            return data;
        } else {
            throw new Exception("pop called on empty stack");
        }

    }
    public static void main(String[] args) {
        Stack<Integer> istack = new Stack<Integer>();
        istack.push(4) ;
        istack.push(5);
        while(true){
            System.out.println(istack.pop());
        }
    }
}

File: linked_lists/stack.py
class Stack:
    class Node:
        def __init__(self, data, nextnode):
            self.data = data
            self.nextnode = nextnode
    def __init__(self):
        self.head = None
    def push(self, data):
        node = self.Node(data, self.head)
        self.head = node
    def pop(self):
        if self.head is not None:
            data = self.head.data
            self.head = self.head.nextnode
            return data
        else:
            raise StopIteration("stack done")

if __name__ == "__main__":
    Hi = Stack()
    Hi.push(1)
    Hi.push(1)
    Hi.push(3)
    print(Hi.pop())
    Hi.push(8)
    Hi.push(10)
    while True:
        print (Hi.pop());
